test/fiber/scheduler.rb

```diff
@@ -96,6 +98,16 @@ def run
     @urgent = nil
   end
 
+  def close
+    self.run
+  ensure
+    @closed = true
```

Comment:
Should we call `freeze` after this, to make sure nothing uses this Scheduler instance anymore?

Reply:
This is just test scheduler, so I'm fine with doing that. I don't think that should be part of the public interface though - is that what you are thinking?

Reply:
I'm just suggesting to add it here, as the test scheduler is also an example for other schedulers.
`freeze` is a just a convenient way to avoid `raise if @closed` in all methods.